### DISCUSSION TITLE
how to extract text from 2 column pdf file

### DISCUSSION THREAD

**Discussion by bugm** [jsvine/pdfplumber#890] [Ask for help with specific PDFs]
Hello guys, suppose I have pdf page looks like
![image](https://github.com/jsvine/pdfplumber/assets/11450151/dd279364-d817-4294-b2ce-ed72805147b8)

I want to extract the text in the order like

**4. Results and discussion
4.1. Experimental setup and simulation environment
To test the performance of the prototype, an experiment setup
for the biaxial-pendulum vibration energy harvester is established
and the schematic is shown in Fig. 6. The experimental setup
mainly includes a six-DOF platform, an inertial measurement unit
4.4. Loaded results from unidirectional excitation
The unidirectional excitation experiments under loaded condi
tions are carried out to examine the loaded performance of the
prototype. Fig. 9 shows the output voltage waveforms of the energy
harvester when the excitation frequency is set to 1 Hz, 1.5 Hz and
2 Hz and the excitation amplitude is set to 0.05 m and 0.07 m,
respectively. The peak values in Fig. 9 (a), (b), and (c) are 1.7 V,10.1 V**

which means extract text for left column top to bottom and then for right column top to bottom. I have tried with page.extract_text() method and did not find the way to get my desired result. I think maybe I can parse it according to the middle x coordinate for page and the x coordinate for each char. Before this I want to know if there is any built-in function in pdfplumber or any other easier way to solve it? 
Thanks a lot!

**Comment by jsvine** [comment]
Hi @bugm, and thanks for your interest in this library. Does this guidance here help?:

- https://github.com/jsvine/pdfplumber/issues/244#issuecomment-670585479
- https://github.com/jsvine/pdfplumber/issues/620#issuecomment-1065021233

**Reply by bugm** [reply]
@jsvine 
Yes,  it helps and thanks for your answer.
By the way , I meet the situation that a pdf page contains both one column line and two columns line. For example, the abstract is one column line and then the text is two columns line. So I have to find the top of the two columns part. 
Now I trying to set different top var and check the 3% middle crop with text extraction on it to see if there's any text or not. Do you have any suggest about it?

**Reply by jsvine** [reply]
Interesting! Another possible approach would be to use `.extract_words(...)`, and to examine the position of words that overlap with the center.

**Reply by bugm** [reply]
fine, I will try it and thanks a lot !
